src/Bindle/ConnectionInfo.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Linq;
+
+namespace Deislabs.Bindle;
+
+public class ConnectionInfo
+{
+    static readonly string[] serverAliases = {
+        "server",
+        "host",
+        "data source",
+        "datasource",
+        "address",
+        "addr",
+        "network address",
+    };
+
+    static readonly string[] sslModeAliases = {
+        "sslmode",
+        "ssl mode"
+    };
+
+    public string? BaseUri;
+
+    public SslMode? SslMode;
+
+    public ConnectionInfo(string connectionString)
+    {
+        BaseUri = GetValue(connectionString, serverAliases);
+
+        try
+        {
+            SslMode = Enum.Parse<SslMode>(GetValue(connectionString, sslModeAliases), true);
+        }
+        catch (Exception)
+        {
+            SslMode = null;
+        }
+    }
+
+    static string GetValue(string connectionString, params string[] keyAliases)
+    {
+        var keyValuePairs = connectionString.Split(';')
```

Comment:
It's arguably irrelevant given the small number of settings but you could just parse into a dictionary once at the start of `new` instead of re-parsing on every `GetValue`.